packages/Dendrogram/src/apps/hierarchical-clustering-sequences-app.ts

```diff
@@ -0,0 +1,64 @@
+import * as ui from 'datagrok-api/ui';
+import * as grok from 'datagrok-api/grok';
+import * as DG from 'datagrok-api/dg';
+
+import {_package} from '../package';
+import {hierarchicalClusteringUI} from '../utils/hierarchical-clustering';
+import {DistanceMetric} from '@datagrok-libraries/bio/src/trees';
+
+export class HierarchicalClusteringSequencesApp {
+  private viewed: boolean = false;
+
+  private _df!: DG.DataFrame;
+  get df(): DG.DataFrame { return this._df; }
+
+  constructor() {}
+
+  async init(df?: DG.DataFrame): Promise<void> {
+    await this.loadData(df);
+  }
+
+  async loadData(df?: DG.DataFrame): Promise<void> {
+    if (!df) {
+      const csv: string = await _package.files.readAsText('data/Prots.csv');
+      df = DG.DataFrame.fromCsv(csv);
+    }
+
+    await this.setData(df);
+  }
+
+  async setData(df: DG.DataFrame) {
+    if (this.viewed) {
+      await this.destroyView();
+      this.viewed = false;
+    }
+
+    this._df = df;
+
+    if (!this.viewed) {
+      await this.buildView();
+      this.viewed = true;
```

Comment:
you don't really need this if, because you'll always get into it
if you have true from the beginning, you'll at first destroy the view and then build new one
if you have false from the beginning, then you'll build the view
so if here is unnecessary

Reply:
true, that bit of code I copied over and didn't pay much attention... thanx!